appengine/myria_web_main.py

```diff
@@ -199,21 +199,24 @@ def get_connection_string(self):
         return connection_string
 
     def base_template_vars(self):
-        try:
-            url = "http://169.254.169.254/latest/meta-data/public-hostname"
-            result = urlfetch.fetch(url).content
-        except:
-            result = "localhost"
-
         if self.app.ssl:
             uri_scheme = "https"
         else:
             uri_scheme = "http"
+
+        try:
+            meta = "http://169.254.169.254/latest/meta-data/public-hostname"
+            hostname = urlfetch.fetch(meta).content
+            jupyterURL = "{s}://{h}:{p}".format(
+                s=uri_scheme, h=hostname, p="8888")
```

Comment:
I think we should be pulling the jupyter host/port from env vars in `app.yaml`, like the MyriaX host/port:

```
env_variables:
  MYRIAX_REST_HOST: localhost
  MYRIAX_REST_PORT: 8753
```
That makes it simple to configure with `myria-cluster`.